src/core/plugins/postmarketos/api.js

```diff
@@ -0,0 +1,102 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const axios = require("axios");
+
+/** @module asteroid_os */
+
+const baseURL = "https://images.postmarketos.org";
+
+const api = axios.create({ baseURL, timeout: 15000 });
+
+/**
+ * get images from api
+ * @param {String} release release
+ * @param {String} user interface
+ * @param {String} device device codename
+ * @returns {Promise<Array<Object>>} images array
+ * @throws {Error} message "no network" if request failed
+ */
+const getImages = (release, ui, device) =>
+  api
+    .get("/bpo/index.json")
+    .then(({ data }) => {
+      const rel = data.releases.find(c => c.name === release);
+      const dev = rel.devices.find(d => d.name.includes(device));
+      const images = dev.interfaces.find(i => i.name === ui).images;
+      // The first two are the latest rootfs and boot image
+      const ts_latest = images[0].timestamp;
+      return images.filter(i => i.timestamp === ts_latest).map(i => ({
+        url: i.url,
+        checksum: {
+          sum: i.sha256, algorithm: "sha256"
+        }
+      }));
+    })
+    .catch(error => {
+      if (error?.response?.status === 404) throw new Error("404");
+      else throw error;
+    });
+
+/**
+ * get releases from api
+ * @param {String} device device codename
+ * @returns {Promise<Array<String>>} releases
+ * @throws {Error} message "unsupported" if 404 not found
+ */
+const getReleases = (device) =>
+  api
+    .get("/bpo/index.json")
+    .then(({ data }) => {
+      let releases = data.releases;
+      return releases.filter(release => release.devices.find(d => d.name.includes(device)))
+        .map(release => release.name);
+    })
+    .catch(error => {
+      if (error?.response?.status === 404) throw new Error("404");
+      else throw error;
+    });
+
+/**
+ * get interfaces from api
+ * @param {String} device device codename
+ * @returns {Promise<Array<String>>} interfaces
+ * @throws {Error} message "unsupported" if 404 not found
+ */
+const getInterfaces = device =>
+  api
+    .get("/bpo/index.json")
+    .then(({ data }) => {
+      let releases = getReleases(device);
+      const devices = data.releases.find(c => c.name === "edge").devices;
+      let interfaces = devices.find(d => d.name.includes(device)).interfaces.map(i => i.name);
+      return interfaces.map(i => {
+        if (i === "phosh") return {value: i, label: "Phosh" }
+        if (i === "plasma-mobile") return {value: i, label: "Plasma Mobile" }
+        if (i === "sxmo-de-sway") return {value: i, label: "SXMO Sway" }
+        return {value: i, label: i }
+      })
```

Comment:
```suggestion
      return interfaces.map(i =>
        (i === "phosh") ? ({value: i, label: "Phosh" }) :
        (i === "plasma-mobile") ? ({value: i, label: "Plasma Mobile" }) :
        (i === "sxmo-de-sway") ? ({value: i, label: "SXMO Sway" }) :
        ({value: i, label: i })
      )
```

also run `npm run lint` over this one again after committing this :]